internal/config/config.go

```diff
@@ -0,0 +1,150 @@
+package config
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+)
+
+// Config struct contains the server configuration
+type Config struct {
+	PermitPasswordLogin bool   `json:"PermitPasswordLogin"`
+	PermitKeyLogin      bool   `json:"PermitKeyLogin"`
+	PermitRootLogin     bool   `json:"PermitRootLogin"`
+	AuthorizedKeysFile  string `json:"AuthorizedKeysFile"`
+	PrivateKeyFile      string `json:"PrivateKeyFile"`
+	EventsLogFile       string `json:"EventsLogFile"`
+	SystemLogFile       string `json:"SystemLogFile"`
+	ListenPort          int    `json:"ListenPort"`
+	ListenAddress       string `json:"ListenAddress"`
+}
+
+// ParseConfig try to open and parse the file at the specified path.
+// If the path is invalid or empty, the function will try to find a config file
+// at the default locations.
+func (c *Config) ParseConfig(path string) error {
+	home, err := os.UserHomeDir()
+
+	if err != nil {
+		return err
+	}
+
+	// Default values if no configuration is provided for them
+	defaultConfigPaths := []string{
+		"/etc/open-bastion/open-bastion-conf.json",
+		home + "/.config/open-bastion/open-bastion-conf.json",
+		home + "/.config/open-bastion-conf.json",
```

Comment:
What do you think of:
* `/etc/open-bastion/config.json`
* `$HOME/.config/open-bastion/config.json`
* `$HOME/.open-bastion/config.json`